File: app/src/main/java/com/riders/thelab/data/remote/api/YoutubeApiService.java
package com.riders.thelab.data.remote.api;

import com.riders.thelab.data.local.model.Video;

import java.util.List;

import io.reactivex.rxjava3.core.Single;
import retrofit2.http.GET;

public interface YoutubeApiService{

    //Method to retrieve the youtube content
    @GET("/florent37/MyYoutube/master/myyoutube.json")
    Single<List<Video>> fetchYoutubeVideos();
}

File: app/src/main/java/com/riders/thelab/data/remote/api/YoutubeApiService.kt
package com.riders.thelab.data.remote.api

import com.riders.thelab.data.local.model.Video

import retrofit2.http.GET

interface YoutubeApiService {

    //Method to retrieve the youtube content
    @GET("/florent37/MyYoutube/master/myyoutube.json")
    suspend fun fetchYoutubeVideos(): List<Video>
}